File: webapi/DAL/Models/Payment.cs
using System;
using System.Collections.Generic;

namespace DAL.Models
{
    public partial class Payment
    {
        public int Id { get; set; }
        public int AgreementId { get; set; }
        public int PaymentMethod { get; set; }
        public DateTime PaymentDueDate { get; set; }
        public string? ChequeNo { get; set; }
        public DateTime? ChequeDueDate { get; set; }
        public int? SenderBankId { get; set; }
        public int? ReceiverBankId { get; set; }
        public int PaymentStatus { get; set; }
        public int Amount { get; set; }
        public string? Description { get; set; }
        public DateTime? PaymentClearanceDate { get; set; }

        public virtual Agreement Agreement { get; set; } = null!;
        public virtual Bank? ReceiverBank { get; set; }
        public virtual Bank? SenderBank { get; set; }
    }
}

File: angular16-boilerplate/src/app/shared/models/payment.ts
export class Payment {
    public id: number;
    public agreementId: number;
    public paymentMethod: number;
    public paymentDueDate: Date;
    public chequeNo?: string | null;
    public chequeDueDate?: Date | null;
    public senderBankId?: number | null;
    public receiverBankId?: number | null;
    public paymentStatus: number;
    public amount: number;
    public description?: string | null;
    public paymentClearanceDate?: Date | null;
    public agreementDates?: string | null;
    public companyName?: string | null;
    public companyBranchOfc?: string | null;
    public receiverBankName?: string | null;
    public senderBankName?: string | null;

    constructor(
        id: number,
        agreementId: number,
        paymentMethod: number,
        paymentDueDate: Date,
        paymentStatus: number,
        amount: number
    ) {
        this.id = id;
        this.agreementId = agreementId;
        this.paymentMethod = paymentMethod;
        this.paymentDueDate = paymentDueDate;
        this.paymentStatus = paymentStatus;
        this.amount = amount;
    }
}